external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala

```diff
@@ -38,6 +40,7 @@ private[kinesis] class KinesisInputDStream[T: ClassTag](
     val endpointUrl: String,
     val regionName: String,
     val initialPositionInStream: InitialPositionInStream,
+    val initialPositionInStreamTimestamp: Date,
```

Comment:
I think there needs to be a better abstraction around the ```initialPositionInStream``` and ```initialPositionInStreamTimestamp``` options. Providing both is redundant as a user would want to specify one or the other. Additionally, if ```initialPositionInStreamTimestamp``` is an optional value then its type should at the very least be ```Option[Date]``` with a default value of ```None```